plugins/reach/admin/scripts/plugin/reach-profile-list-action.phtml

```diff
@@ -0,0 +1,407 @@
+
+		dialogDiv.dialog({
+			bgiframe: true,
+			modal: true,
+			resizable: false,
+			width: 600,
+			height: 600,
+			title: '<?php echo $this->translate('Reach Profile'); ?>',
+			buttons: {
+				'<?php echo $this->translate('Save'); ?>': function() {
+					jQuery('.ui-widget-overlay').css('z-index', 1003 );
+					jQuery('#loader').show();
+					loadArrayObjectsData();
+					convertDates();
+					jQuery(this).find('#frmReachProfileConfigure').submit();
+				},
+				'<?php echo $this->translate('Close'); ?>': function() {
+					jQuery(this).dialog('close');
+				}
+			},
+			close: function() {
+				jQuery(this).dialog('destroy').remove();
+			}
+		});
+
+		dialogDiv.load(
+			url,
+			null,
+			function() {
+				jQuery(this).find('#frmReachProfileConfigure').ajaxForm({
+					success: loadReachProfileSuccess,
+					iframe: true
+				});
+			}
+		);
+	}
+
+	function loadArrayObjectsData() {
+		//remove all template field
+		$('fieldset[id^="reachProfileRuleTemplate"]').remove();
+		setRuleData();
+		//remove all template field
+		$('fieldset[id^="DictionaryTemplate"]').remove();
+		setDictionaryData();
+	}
+    
+	function setRuleData() {
+		var rules = [];
+		var currentIndex = 0;
+		$('[description^=AutomaticAdminConsoleRule]').each(function(idx){
+			var newRule= buildOutputParams($( this ).attr("id"), currentIndex);
+			rules.push(newRule);
+		});
+
+		jQuery('#ReachProfileRules').val(JSON.stringify(rules));
+	}
+
+    function buildOutputParams(paramName, index) {
+	    var param = {};
+        jQuery('#' + paramName).find('input,select').each(function(idx) {
+            var type = $( this ).attr("type");
+            var name = $( this ).attr("name");
+            var val = $( this ).val();
+            if (type == 'hidden')
+                return;
+            param[name] = val;
+            $(this).remove();
+        });
+        return param;
+    }
+
+	function loadReachProfileSuccess(html) {
+		if (jQuery(html).filter('#frmReachProfileConfigure').hasClass('valid'))
+		{
+			dialogDiv.dialog('close');
+			if (jQuery('#frmPaginator1').length){
+				jQuery('#frmPaginator1').submit();
+			}
+			else {
+				jQuery('#frmPartnerIdFilter').submit();
+			}
+		}
+		jQuery('.ui-widget-overlay').css('z-index', 1000 );
+		jQuery('#loader').hide();
+		jQuery('#frmReachProfileConfigure').replaceWith(html);
+		jQuery('#frmReachProfileConfigure').ajaxForm({
+			success: loadReachProfileSuccess,
+			iframe: true
+		});
+	}
+
+	function convertDates()
+	{
+		var f = jQuery('#frmReachProfileConfigure');
+		f.find( "input[innertype=DateElement]").each(function(idx) {
+			var val = $(this).val();
+			if (isDatesFormat(val)) {
+				$(this).val(getEpochTime(val));
+				return;
+			}
+			var id = this.id;
+			if (val != 'N/A' && epocToMonthsField.some(function(v) { return id.indexOf(v) >= 0;}))
+				$(this).val(val * 2592000); // 60*60*24*30 - set relative in seconds from relative months
+		});
+	}
+
+	function isDatesFormat(val) {
```

Comment:
Better use Date() class for convert and validate
if(!isNaN(Date.parse(val)) 

Will do the same